### PR TITLE
Use : as seccomp security option operator for Docker 1.10

### PR DESCRIPTION
**Pull request by sttts**
Compare https://docs.docker.com/v1.11/engine/reference/run/ and https://docs.docker.com/v1.10/engine/reference/run/

Fixes https://github.com/kubernetes/kubernetes/issues/28107

### CODE REVIEW

**Comment by ncdc** [comment]
@eparis @erictune @pmorie @jfrazelle this fixes seccomp with Docker 1.10. I've added the cherrypick-candidate label as this would be nice to pull into the 1.3 branch, since we added this feature for 1.3.

**Comment by ncdc** [comment]
cc @vishh @dchen1107 @kubernetes/rh-cluster-infra

**Comment by ncdc** [comment]
@sttts looks like this change broke TestSeccompLocalhostProfileIsLoaded

**Comment by sttts** [comment]
@ncdc will take a look

File: pkg/kubelet/dockertools/docker_manager.go [hunk 1]
```diff
@@ -993,19 +993,26 @@ func (dm *DockerManager) getSecurityOpt(pod *api.Pod, ctrName string) ([]string,
 		return nil, nil
 	}
 
+	// in Docker 1.10 the operator is :, for anything later it's =
+	optFmt := "%s=%s"
+	if result <= 0 {
```
Comment (aveshagarwal) [inline pkg/kubelet/dockertools/docker_manager.go]: Not sure I am missing something, but above you return when result < 0. Now you are checking result <=0. Do you mean just result = 0 here?

Comment (sttts) [inline pkg/kubelet/dockertools/docker_manager.go]: Both are equivalent due to the check above. Can change it into == 0, makes
it clearer.

File: pkg/kubelet/dockertools/docker_manager.go [hunk 2]
```diff
@@ -993,19 +993,26 @@ func (dm *DockerManager) getSecurityOpt(pod *api.Pod, ctrName string) ([]string,
...
 		}
 	}
 
 	if profile == "unconfined" {
 		// return early the default
-		return defaultSecurityOpt, nil
+		return defaultSecurityOpts, nil
 	}
 
 	if profile == "docker/default" {
```
Comment (aveshagarwal) [inline pkg/kubelet/dockertools/docker_manager.go]: Although this change is not part of this PR,  below it return nil, wonder why not return defaultSecurityOpts?

File: pkg/kubelet/dockertools/docker_manager.go (hunk 1 above)
Comment (sttts) [inline pkg/kubelet/dockertools/docker_manager.go]: ptal.

File: pkg/kubelet/dockertools/docker_manager.go (hunk 2 above)
Comment (sttts) [inline pkg/kubelet/dockertools/docker_manager.go]: Docker default is not unconfined, but something which fits to the selected capabilities. Kubernetes chose another default than Docker here.

**Comment by ncdc** [comment]
LGTM

**Comment by wmengRH** [comment]
@sttts works fine with docker 1.10 now. Thanks.
docker-common-1.10.3-44.el7.x86_64

**Comment by erictune** [comment]
fyi @timstclair
